app/controllers/sitemap_controller.rb

```diff
@@ -0,0 +1,55 @@
+class SitemapController < ActionController::Metal
+
+  include AbstractController::Rendering
+  include ActionController::MimeResponds
+  include ActionController::DataStreaming
+  include ActionController::RackDelegation
+  include ActionController::Rescue
+  include ActionController::Head
+
+  def sitemap
+    render_site_map
+  end
+
+  private
+
+    def render_site_map
+      community = request.env[:current_marketplace]
+      if community.deleted?
+        head :not_found
+        return
+      end
+      if community.private?
+        head :forbidden
+        return
+      end
+      respond_to do |format|
+
+        data = Rails.cache.fetch(community.id, expires_in: 24.hours) do
```

Comment:
It would be better to have some prefix to the cache key. Something like `"sitemaps/#{community.id}"` would be great!